metron-platform/metron-common/src/main/java/org/apache/metron/common/configuration/ConfigurationsUtils.java

```diff
@@ -156,7 +156,7 @@ public static void writeConfigToZookeeper(ConfigurationType configType,Optional<
   }
 
   private static String getConfigZKPath(ConfigurationType configType, Optional<String> configName) {
-    String pathSuffix = configName.isPresent() && configType != GLOBAL ? "/" + configName : "";
+    String pathSuffix = configName.isPresent() && configType != GLOBAL ? "/" + configName.get() : "";
```

Comment:
how did this ever work before?  I'm surprised we didn't get the wrong result because configName is optional.

Reply:
Yeah, this confused me too.  There are a lot of paths through ConfigurationUtils.  Some paths worked, others didn't.

Reply:
We need to refactor `ConfigurationUtils` so badly that it hurts.

Reply:
I think that's because we handle global config in a special way. This whole class needs refactored, and I'd have done more with the original patch PR had it not blown up scope.

Reply:
Yeah, this is not one of those things that can be done as a side-effect of another PR.  I considered refactoring it as part of the zookeeper refactoring that I did earlier.  It needs more direct attention and refactoring.